Sources/Utility/ArgumentParser.swift

```diff
@@ -382,12 +361,94 @@ public final class ArgumentParser {
         }
     }
 
+    /// An enum representing way to parse argument values.
+    public enum ParsingStrategy {
+        case one
+        case multiple
+        case remainder
+        
+        func parse(_ parser: inout Parser, kind: ArgumentKind.Type) throws -> [ArgumentKind] {
+            var values: [ArgumentKind] = []
+            
+            switch self {
+            case .one:
+                var parserProtocol = parser as ArgumentParserProtocol
+                try values.append(kind.init(parser: &parserProtocol))
+                parser = parserProtocol as! Parser
+            case .multiple:
+                while true {
+                    let previousIterator = parser.argumentsIterator
+                    
+                    guard let nextArgument = try? parser.next() else {
+                        break
+                    }
+                    
+                    if isPositional(argument: nextArgument) {
+                        try values.append(kind.parse(argument: nextArgument))
+                    } else {
+                        parser.argumentsIterator = previousIterator
```

Comment:
It is required: if we have reached a non-positional argument, we need back-track to the state of the iterator before it, so that the top-level parser can parse that non-positional argument.